operator/pkg/s3.go

```diff
@@ -37,11 +38,20 @@ func (o *Operator) getBatchFromS3(batchURL string, expectedMerkleRoot [32]byte)
 		}
 	}(resp.Body)
 
-	batchBytes, err := io.ReadAll(resp.Body)
+	// Use io.LimitReader to limit the size of the response body
+	// This is to prevent the operator from downloading a larger than expected file
+	// + 1 is added to the contentLength to check if the response body is larger than expected
+	reader := io.LimitedReader{R: resp.Body, N: contentLength + 1}
+	batchBytes, err := io.ReadAll(&reader)
 	if err != nil {
 		return nil, err
 	}
 
+	// Check if the response body is larger than expected
+	if reader.N == 0 {
```

Comment:
in the [io.LimitedReader docs](https://pkg.go.dev/io#LimitedReader) I found
> Read returns EOF when N <= 0 or when the underlying R returns EOF. 

should the condition be changed to 
```suggestion
	if reader.N <= 0 {
```
?

Reply:
Seems like it should ne changed, yes